tests/providers/test_alibaba_provider.py

```
@@ -91,3 +99,26 @@ def test_alibaba_provider_custom_base_url():
     provider = AlibabaProvider(api_key='test-key', base_url='https://custom.endpoint.com/v1')
     assert provider.base_url == 'https://custom.endpoint.com/v1'
     assert str(provider.client.base_url).rstrip('/') == 'https://custom.endpoint.com/v1'
+
+
+async def test_alibaba_document_input_not_supported(allow_model_requests: None):
+    provider = AlibabaProvider(api_key='test-key')
+    model = OpenAIChatModel(model_name='qwen-max', provider=provider)
+    agent = Agent(model)
+
+    with pytest.raises(UserError, match='alibaba.*does not support document input'):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)